### PR TITLE
Feature/article content

### PR DESCRIPTION
**Pull request by brunopintos**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,8 @@
   },
   "name": "workspace-library",
   "dependencies": {
+    "@types/styled-components": "^5.0.1",
+    "react-native-webview-quilljs": "^2.0.0",
```
Comment (DogChef) [inline package.json:28]: Root package.json can't have any dependency, you can check that WebViewQuillJS was added as a package folder and we want all our libraries in node_modules. Please remove this and add them on the package.json at mobile.

You can do this from the root folder like so:

` bash ./workspace.sh mobile add react-native-webview-quilljs`
`bash ./workspace.sh mobile add @types/styled-components -D`

Note that I used a -D on @types/styled-components since its a devDependency

File: packages/mobile/components/ArticleContent.tsx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+import { useState, useEffect } from "react";
+
+import styled from "styled-components";
+import { ActivityIndicator } from "react-native-paper";
+import { View, Text, SafeAreaView, Button } from "react-native";
+import moment from "moment";
+import Breadcrumbs from "./Breadcrumbs";
+import RichTextEditor from "./RichTextEditor";
+
+import ARTICLES_QUERY from "../../kbcore/dist/queries/ARTICLES_QUERY";
```
Comment (DogChef) [inline packages/mobile/components/ArticleContent.tsx:12]: Remove any content related to the backend

File: packages/mobile/components/ArticleContent.tsx [hunk 3]
```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import { useState, useEffect } from "react";
+
+import styled from "styled-components";
+import { ActivityIndicator } from "react-native-paper";
+import { View, Text, SafeAreaView, Button } from "react-native";
+import moment from "moment";
+import Breadcrumbs from "./Breadcrumbs";
+import RichTextEditor from "./RichTextEditor";
```
Comment (DogChef) [inline packages/mobile/components/ArticleContent.tsx:10]: Let's not copy the name of RichTextEditor since that's already taken, how does ArticleEditor sound?

File: packages/mobile/components/ArticleContent.tsx [hunk 4]
```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import { useQuery, useMutation } from "@apollo/react-hooks";
```
Comment (DogChef) [inline packages/mobile/components/ArticleContent.tsx:2]: On a note, you'll want to remove this and the queries imported below, we're going to work with the hooks generated by graphql-codegen

File: packages/mobile/components/Breadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  <StyledText>{title}</StyledText>
+);
+
+const BreadcrumbSeparator = ({ separator }: { separator: string }) => (
+  <StyledTextSeparator>{separator}</StyledTextSeparator>
+);
+
+const BreadcrumbCollapser = () => <StyledText>...</StyledText>;
+
+const Breadcrumbs = ({
```
Comment (DogChef) [inline packages/mobile/components/Breadcrumbs.tsx:38]: Great job on deconstructuring props!, as a suggestoin I'd say create an interface with the types so it improves readability.

````
interface Props {
    separator: string;
    items: { title: string }[];
}

const Breadcrumbs = ({ separator, items }: Props) => {
.
.
.
````

File: packages/mobile/components/Breadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  separator,
+  items,
+}: {
+  separator: string;
+  items: { title: string }[];
+}) => {
+  let breadcrumbs = items.map(({ title }) => <BreadcrumbItem title={title} />);
+
+  const max = 4;
```
Comment (DogChef) [inline packages/mobile/components/Breadcrumbs.tsx:47]: Let's define this outside our component

File: packages/mobile/components/Breadcrumbs.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
```
Comment (DogChef) [inline packages/mobile/components/Breadcrumbs.tsx:1]: This component belongs to another functionality, please remove it from this branch

File: packages/mobile/components/RichTextEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  padding: 5px;
+`;
+
+const StyledView = styled(View)`
+  flex: 1;
+  padding: 10px;
+  margin-bottom: 15px;
+`;
+
+export default function RichTextEditor({
```
Comment (DogChef) [inline packages/mobile/components/RichTextEditor.tsx:17]: Same note with the interface

File: packages/mobile/components/RichTextEditor.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  content,
+  onSave,
+}: {
+  content: String;
+  onSave: any;
```
Comment (DogChef) [inline packages/mobile/components/RichTextEditor.tsx:22]: Let's try to avoid using any if possible

File: packages/mobile/navigation/AppNavigator.tsx [hunk 10]
```diff
@@ -2,49 +2,84 @@ import React from "react";
...
-        <Button
+        <IconButton
           style={{ minWidth: 0, left: 4 }}
           icon="menu"
           onPress={() => navigation.toggleDrawer()}
         />
       ),
-      headerTitleContainerStyle: { left: 44 }
+      headerTitleContainerStyle: {
+        left: 4,
```
Comment (DogChef) [inline packages/mobile/navigation/AppNavigator.tsx:41]: This is intended to be 44, why the change?

Comment (brunopintos) [reply packages/mobile/navigation/AppNavigator.tsx:41]: Why 44? To center the title?

File: packages/mobile/navigation/AppNavigator.tsx [hunk 11]
```diff
@@ -2,49 +2,84 @@ import React from "react";
...
-  ScrollView,
-  StyleSheet,
-  Text,
-  TouchableOpacity,
-  View
-} from "react-native";
-import { Button } from "react-native-paper";
+import { IconButton } from "react-native-paper";
+import { View } from "react-native";
+import ArticleContent from "../components/ArticleContent";
```
Comment (DogChef) [inline packages/mobile/navigation/AppNavigator.tsx:7]: Dont add components straight to the navigator, we should have an article screen that uses the article editor

File: packages/mobile/navigation/AppNavigator.tsx [hunk 12]
```diff
@@ -1,42 +1,45 @@
 import React from "react";
 import HomeScreen from "../screens/Home";
+import ArticleScreen from "mobile/screens/Article";
```
Comment (DogChef) [inline packages/mobile/navigation/AppNavigator.tsx:3]: "../screens/Article"
